### ISSUE TITLE
Impossible to install 2 awx on the same cluster in 2 different namespace with HELM

### ISSUE THREAD

**Issue by milodanlso** [ansible/tower-operator#1290]
### Please confirm the following

- [X] I agree to follow this project's [code of conduct](https://docs.ansible.com/ansible/latest/community/code_of_conduct.html).
- [X] I have checked the [current issues](https://github.com/ansible/awx-operator/issues) for duplicates.
- [X] I understand that the AWX Operator is open source software provided for free and that I might not receive a timely response.

### Bug Summary

I have 2 namespace : infra-int and infra-dev. AWX run in infra-dev.

When I try to install with helm the second AWX in infra-int I get the following message :

Error: INSTALLATION FAILED: rendered manifests contain a resource that already exists. Unable to continue with install: ClusterRole "awx-operator-metrics-reader" in namespace "" exists and cannot be imported into the current release: invalid ownership metadata; annotation validation error: key "[meta.helm.sh/release-namespace](http://meta.helm.sh/release-namespace)" must equal "infra-int": current value is "infra-dev"

### AWX Operator version

1.1.1

### AWX version

21.10

### Kubernetes platform

kubernetes

### Kubernetes/Platform version

1.24.9

### Modifications

no

### Steps to reproduce

Install AWX in the first namespace (example: infra-dev). Then install the second AWX in another namespace (e.g. infra-int).
The message in the bug summary appears.
I asked about this on the forum and it seems that this type of installation is not a problem using `make deploy` or the Kustomization approach
So it seems to be related to Helm only (I don't have enough expertise to say more, sorry)

### Expected results

Have one awx under infra-int and another under infra-dev

### Actual results

I have the message :

Error: INSTALLATION FAILED: rendered manifests contain a resource that already exists. Unable to continue with install: ClusterRole "awx-operator-metrics-reader" in namespace "" exists and cannot be imported into the current release: invalid ownership metadata; annotation validation error: key "[meta.helm.sh/release-namespace](http://meta.helm.sh/release-namespace)" must equal "infra-int": current value is "infra-dev"

**Comment by rooftopcellist** [comment]
FYI, I can reproduce that issue locally with multiple helm deployments in different namespaces.  

However, I don't run into that issue using `make deploy` or the Kustomization approach:
* https://github.com/ansible/awx-operator#basic-install

This would be a good issue for community folks using the helm operator to help out with.  I'll add the `help_wanted` label.

**Comment by miles-w-3** [comment]
This is an interesting problem, `ClusterRoles` work across namespaces so it makes sense that the collision would happen there. In order to make this work, we'd need to remove all annotations that are specific to a given chart installation (release name and namespace) and then helm might allow it. However, helm does store metadata for all of its released resources that kustomize does not, so I'm not sure we would be able to eliminate all claims to it.

In some respects, this is a good warning from helm because you have 2 installations creating the same resource. I think one of the dangers of two installs managing the same resource is that if one installation gets updated, it impacts these shared resources and therefore the other install as well. Basically, I think the same fundamental conflict exists in the kustomize version, but kustomize can patch and won't make a fuss about it at the top level. I think the preferred approach could be do add a flag that disables cluster-wide resource installations for a given helm install, in addition to assessing whether this ClusterRole needs to have cluster scope.

**Comment by grimlokason** [comment]
Hello,

ClusterRoles should be reassigned as Roles since the awx-operator is namespaced  ( https://sdk.operatorframework.io/docs/building-operators/golang/operator-scope/#changing-the-permissions-to-namespaced ) 
Is there some issue in doing it ?

Once done, we will be able to deploy the operator in different namespace,

Best regards.

**Comment by jeremych1000** [comment]
I'm confused as well. awx-operator says it's now namespaced, but still creates clusterroles in the helm chart? I can skip the CRDs with --skip-crds, but how can you create clusterroles in a namespaced chart?!

**Comment by jeremych1000** [comment]
Any update on this please?

Context - we are running a multi-tenant Kubernetes cluster i.e. namespace as a service. One of the tenants wants to install awx-operator, and we've had to manually add the CRDs for them.

The helm chart should be creating roles, not clusterroles, especially if the paradigm is to install a namespace scoped operator.

cc. @adamjohnson01

**Comment by miles-w-3** [comment]
To be clear, this issue isn't specific to the helm chart. Except for the AWX resource itself, the helm chart pulls all of the other resources from the kustomize-based operator resources

**Comment by grimlokason** [comment]
Hello, 
The issue is from those 3 files : 

```
[awx-operator]$ cat Chart.yaml  |grep version
version: 2.1.0
[awx-operator]$ grep -Hri clusterrole *
templates/clusterrolebinding-awx-operator-proxy-rolebinding.yaml:kind: ClusterRoleBinding
templates/clusterrolebinding-awx-operator-proxy-rolebinding.yaml:  kind: ClusterRole
templates/clusterrole-awx-operator-proxy-role.yaml:kind: ClusterRole
templates/clusterrole-awx-operator-metrics-reader.yaml:kind: ClusterRole
```

It work if we change clusterrole to role , clusterrolebinding to rolebinding and delete the clusterrole-awx-operator-metrics-reader.yaml since the nonressourcesurl are only for clusterrole. 
I don't know what the clusterrole-awx-operator-metrics-reader.yam is for, and how we can change it to namespaced so i don't propose a PR.


Best regards.

**Comment by miles-w-3** [comment]
Yes but this isn't a standard helm chart. These files are pulled from the kustomize directories and put into the helm template folder. So the change goes there, and I'm not sure if there are any additional considerations with the operator logic

**Comment by rverhe** [comment]
Any updates on this? Having the same issue.
Or a workaround?

**Comment by grimlokason** [comment]
the only workaround i've found is here : 

> Hello, The issue is from those 3 files :
> 
> ```
> [awx-operator]$ cat Chart.yaml  |grep version
> version: 2.1.0
> [awx-operator]$ grep -Hri clusterrole *
> templates/clusterrolebinding-awx-operator-proxy-rolebinding.yaml:kind: ClusterRoleBinding
> templates/clusterrolebinding-awx-operator-proxy-rolebinding.yaml:  kind: ClusterRole
> templates/clusterrole-awx-operator-proxy-role.yaml:kind: ClusterRole
> templates/clusterrole-awx-operator-metrics-reader.yaml:kind: ClusterRole
> ```
> 
> It work if we change clusterrole to role , clusterrolebinding to rolebinding and delete the clusterrole-awx-operator-metrics-reader.yaml since the nonressourcesurl are only for clusterrole. I don't know what the clusterrole-awx-operator-metrics-reader.yam is for, and how we can change it to namespaced so i don't propose a PR.
> 
> Best regards.

It also seems that there is some upgrade in the helm chart : https://github.com/ansible/awx-operator/pull/1159 but i haven't tested it yet. ( probably tomorrow )
